core/src/main/java/org/apache/calcite/sql/fun/SqlCastFunction.java

```diff
@@ -93,6 +95,29 @@ public SqlCastFunction() {
 
   //~ Methods ----------------------------------------------------------------
 
+  @Override public void validateCall(final SqlCall call, final SqlValidator validator,
+      final SqlValidatorScope scope,
+      final SqlValidatorScope operandScope) {
+    try {
+      super.validateCall(call, validator, scope, operandScope);
+    } catch (UnsupportedOperationException ex) {
+      assert call.operandCount() == 2;
+      throw validator.newValidationError(call,
+          RESOURCE.cannotCastToType(call.operand(1).toString()));
```

Comment:
Should the original `UnsupportedOperationException` be kept as a `cause` of the exception?

Reply:
no

Reply:
The doc string of `SqlValidatorException` [link](https://github.com/apache/calcite/blob/4d413bb21fb0a18882c5066f1d75a02d5b021bac/core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorException.java#L36) says that it is a checked exception. I'm guessing Julian is saying that wrapping `UnsupportedOperationException`, a runtime exception, in a checked exception does not make sense?

Reply:
As I said in the JIRA case, this is not an exceptional case, and UnsupportedOperationException should not be thrown anywhere along this code path.

Don't approach this task with the framing 'how do we handle this exception?', but rather 'how do we make the validator do its job and clearly identify to the user types that are invalid?'.